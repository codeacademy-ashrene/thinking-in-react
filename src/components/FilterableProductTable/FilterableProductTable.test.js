import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import FilterableProductTable from '.';

describe('FilterableProductTable', () => {
  const productArray = [
    {category: "Sporting Goods", price: "$49.99", stocked: true, name: "Football"},
    {category: "Sporting Goods", price: "$9.99", stocked: true, name: "Baseball"},
    {category: "Sporting Goods", price: "$29.99", stocked: false, name: "Basketball"},
    {category: "Electronics", price: "$99.99", stocked: true, name: "iPod Touch"},
    {category: "Electronics", price: "$399.99", stocked: false, name: "iPhone 5"},
    {category: "Electronics", price: "$199.99", stocked: true, name: "Nexus 7"}
  ];
  it('renders without crashing for stocked and unstocked products and no filter text', () => {
    const filterableProductTable = renderer.create(<FilterableProductTable products={productArray} />).toJSON();
    expect(filterableProductTable).toMatchSnapshot();
  });
  it('should change state when value is passed to checkbox', () => {
    const wrapper = mount(<FilterableProductTable products={productArray} />);
    expect(wrapper.instance().state.inStockOnly).toEqual(false);
    wrapper.find('.checkBox').simulate('change', {target: {checked: true}});
    expect(wrapper.instance().state.inStockOnly).toEqual(true);
  });
  it('should change state when value is passed to search box', () => {
    const wrapper = mount(<FilterableProductTable products={productArray} />);
    expect(wrapper.instance().state.filterText).toEqual('');
    wrapper.find('.search').simulate('change', {target: {value: 'Football'}});
    expect(wrapper.instance().state.filterText).toEqual('Football');
  })
});
