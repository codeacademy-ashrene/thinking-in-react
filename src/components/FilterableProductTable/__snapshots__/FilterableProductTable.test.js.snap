// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterableProductTable renders without crashing for stocked and unstocked products and no filter text 1`] = `
<div>
  <form>
    <input
      className="search"
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value=""
    />
    <input
      checked={false}
      className="checkBox"
      onChange={[Function]}
      type="checkbox"
    />
     
    Only show products in stock
  </form>
  <table>
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          Price
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <th>
          Sporting Goods
        </th>
      </tr>
      <tr>
        <td>
          <span>
            Football
          </span>
        </td>
        <td>
          $49.99
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Baseball
          </span>
        </td>
        <td>
          $9.99
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={
              Object {
                "color": "red",
              }
            }
          >
            Basketball
          </span>
        </td>
        <td>
          $29.99
        </td>
      </tr>
      <tr>
        <th>
          Electronics
        </th>
      </tr>
      <tr>
        <td>
          <span>
            iPod Touch
          </span>
        </td>
        <td>
          $99.99
        </td>
      </tr>
      <tr>
        <td>
          <span
            style={
              Object {
                "color": "red",
              }
            }
          >
            iPhone 5
          </span>
        </td>
        <td>
          $399.99
        </td>
      </tr>
      <tr>
        <td>
          <span>
            Nexus 7
          </span>
        </td>
        <td>
          $199.99
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
